eventlog/api.go

```diff
@@ -202,3 +206,136 @@ func (c *Client) Search(req *SearchRequest) (*SearchResponse, error) {
 	}
 	return reply, nil
 }
+
+// StreamHandler is the signature of the handler used when streaming events.
+type StreamHandler func(event Event, blockID []byte, err error)
+
+// Close closes all the websocket connections.
+func (c *Client) Close() error {
+	err := c.ByzCoin.Close()
+	if err2 := c.sc.Close(); err != nil {
+		err = err2
+	}
+	if err2 := c.c.Close(); err != nil {
+		err = err2
+	}
+	return err
+}
+
+// StreamEvents is a blocking call where it calls the handler on even new event
+// until the connection is closed or the server stops.
+func (c *Client) StreamEvents(handler StreamHandler) error {
+	h := func(resp byzcoin.StreamingResponse, err error) {
+		if err != nil {
+			handler(Event{}, nil, err)
+			return
+		}
+		// don't need to handle error because it's given to the handler
+		_ = handleBlocks(handler, resp.Block)
+	}
+	// the following blocks
+	return c.ByzCoin.StreamTransactions(h)
+}
+
+// StreamEventsFrom is a blocking call where it calls the handler on even new
+// event from (inclusive) the given block ID until the connection is closed or
+// the server stops.
+func (c *Client) StreamEventsFrom(handler StreamHandler, id []byte) error {
```

Comment:
Should `id` be of type skipchain.SkipBlockID ?

Reply:
It is for convenience, I didn't want to expose the user to the skipchain package. It's the same reason why the handler which you implement uses a byte slice `type StreamHandler func(event Event, blockID []byte, err error)`.